cpp/include/cudf/utilities/traits.hpp

```diff
@@ -502,4 +525,26 @@ constexpr inline bool is_nested(data_type type)
 }
 
 /** @} */
+
+template <typename From, typename To>
+struct is_convertible : std::is_convertible<From, To> {
+};
+
+// This will ensure that timestamps can be promoted to a higher precision. Presently, they can't
+// do that due to nvcc/gcc compiler issues
+template <typename Duration1, typename Duration2>
+struct is_convertible<cudf::detail::timestamp<Duration1>, cudf::detail::timestamp<Duration2>>
+  : std::is_convertible<typename cudf::detail::time_point<Duration1>::duration,
+                        typename cudf::detail::time_point<Duration2>::duration> {
+};
+
 }  // namespace cudf
+
+namespace std {
+// This specialization is required due to an nvcc bug. This works in 10.1.
+// We can remove this once the build pipeline removes 10.0.
+template <typename Duration>
+struct is_trivially_copyable<cudf::detail::timestamp<Duration>>
```

Comment:
i'll remove this before the commit; i'm keeping this only to get the build pipeline to pass.

Reply:
Is this left in so the CUDA 10.0 CI job passes? Are we expecting to remove CUDA 10.0 support before merging this PR? If not, should we guard this definition with an `#ifdef` so it's only include when the CUDA version is strictly 10.0?

Reply:
i have now guarded this such that this specialization is applicable *only* for 10.0 builds.

Reply:
> I would not take this out until we are well past 10.0, just in case someone chooses to keep using CUDA 10.0 "at their own risk". It's one thing to not support a CUDA version, but to knowingly block it is another...

It is undefined behavior to include this specialization. We shouldn't be including it in a released build whatsoever.

Reply:
there is a guard to enable this *only* for cuda 10.0. if this is removed, then 10.0 ci builds. i'll remove this if 10.0 support is removed from ci.

Reply:
it looks like we are no longer building on 10.0. i'll remove this code block and check to be sure.